packages/design-system/src/components/primitives/dnd/dom-utils.ts

```diff
@@ -0,0 +1,70 @@
+import {
+  type Rect,
+  type ChildrenOrientation,
+  getRectsOrientation,
+} from "./geometry-utils";
+
+export type UsableElement = HTMLElement | SVGElement;
+
+// Many DOM APIs produce values of type Element,
+// which lacks certain methods that we often need.
+// It's also usually safe to assume that the value is in fact HTMLElement or SVGElement.
+// We use this helper to convince TypeScript that the value is a UsableElement.
+export const toUseableElement = (
```

Comment:
if its only for typescript, there should be a way to do this in types only, without generating runtime code, wdyt? or do we actually need the runtime checks?

Reply:
I think everywhere where I use this helper, the check always passes. But I'm not 100% sure. Also, not sure how to do this using only types.

Reply:
I guess you could use an assertion like `as HTMLElement` if you only need type check

Reply:
Ah, right. I'm used to Flow that wouldn't allow that because it's unsafe:

https://www.typescriptlang.org/play?#code/MYGwhgzhAEAqCmEAu0DeBfAUJ0kYOQGEALASxABNp4APJeAOwv0RVU2mgDcwQBXeAC5oDPgFsARvABOAbg7RgAewbJpfYEiXSAFAEo0CzhD4AHGfvmdOSMhAB0PfvGgBeaAEYr0LFhwrkaHpA9wZ4AHc4VktsZVUUJwFhUUkZN2gdYJRIKKIySj1HXgEgA

But for our case it's probably better to use `as HTMLElement` indeed.